hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -818,3 +818,77 @@ def test_advanced_integer_index_can_generate_any_pattern(shape, data):
         lambda index: np.all(target == x[index]),
         settings(max_examples=10 ** 6),
     )
+
+
+@pytest.mark.parametrize(
+    "condition",
+    [
+        lambda ix: Ellipsis in ix,
+        lambda ix: Ellipsis not in ix,
+        lambda ix: np.newaxis in ix,
+        lambda ix: np.newaxis not in ix,
+    ],
+)
+def test_test_basic_indices_options(condition):
+    indexers = nps.array_shapes(min_dims=2, max_dims=32).flatmap(
```

Comment:
```suggestion
    indexers = nps.array_shapes(min_dims=0, max_dims=32).flatmap(
```

^ works once you remove the shape-length check